golem_messages/datastructures/tasks.py

```diff
@@ -0,0 +1,142 @@
+import functools
+import hashlib
+import logging
+import time
+
+from golem_messages import cryptography
+from golem_messages import datastructures
+from golem_messages import exceptions
+from golem_messages import serializer
+from golem_messages import validators
+from golem_messages.datastructures import masking
+from golem_messages.datastructures import p2p as dt_p2p
+
+
+logger = logging.getLogger(__name__)
+
+
+def _fail_if(field_name, value, check, fail_msg):
+    if not check(value):
+        raise exceptions.FieldError(
+            fail_msg,
+            field=field_name,
+            value=value,
+        )
+
+
+class TaskHeader(datastructures.Container):
+    """
+    Task header describes general information about task as an request and
+    is propagated in the network as an offer for computing nodes
+    """
+
+    __slots__ = {
+        'mask': (
+            functools.partial(
+                _fail_if,
+                check=lambda x: isinstance(x, bytes),
+                fail_msg="Should be bytes",
+            ),
+        ),
+        'timestamp': (
+            validators.validate_float,
+        ),
+        'signature': (
+            functools.partial(
+                _fail_if,
+                check=lambda x: isinstance(x, bytes),
+                fail_msg="Should be bytes",
+            ),
+        ),
+        'task_id': (validators.validate_varchar128, ),
+        'task_owner': (
+            functools.partial(
+                _fail_if,
+                check=lambda x: isinstance(x, (dict, dt_p2p.Node)),
+                fail_msg="Should be a dict or Node",
+            ),
+        ),
+        'deadline': (
+            validators.validate_float,
+            functools.partial(
+                _fail_if,
+                check=lambda x: x > time.time(),
+                fail_msg="Deadline already passed",
+            ),
+        ),
+        'subtask_timeout': (
+            validators.validate_float,
```

Comment:
I think that timeouts should be ints. See my comments at https://github.com/golemfactory/golem/pull/3655